nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/JMSPublisher.java

```diff
@@ -89,11 +89,14 @@ void setMessageHeaderAndProperties(final Message message, final Map<String, Stri
             for (Entry<String, String> entry : flowFileAttributesToSend.entrySet()) {
                 try {
                     if (entry.getKey().equals(JmsHeaders.DELIVERY_MODE)) {
-                        message.setJMSDeliveryMode(Integer.parseInt(entry.getValue()));
+                        this.jmsTemplate.setExplicitQosEnabled(true);
```

Comment:
are setting these values explicitly and without any configurability going to limit usage of these processors for certain JMS brokers?

Reply:
No, it should not. The only changes is setting these values through the JmsTemplate vice the Message itself due to stated limitation of the API. The net effect is the same.